appmanager/franca/appmgr_core.fidl

```diff
@@ -33,42 +32,33 @@ interface AppManager_Core
 			AppInfo info
 		}
 	}
-	
-	
-	
-	<** 
-		@description : this gives the current active application (the one which is currently being displayed)
-	**>
-		 
-	attribute String CurrentActiveApp readonly
-	
+
 	<**
 		@description : launch an app using the app_id. App manager resolves the app_id to an entry point and launches it.
 	**>
-	
 	method LaunchApp
 	{
 		in
 		{
 			String app_id
 		}
-		
 	}
 	
-	
 	<**
 		@description : This should signal the addition or deletion of new apps, change in capability of apps to handle mimes etc. It si adviced to call GetAppMimeTable and GetInstalledApps as needed when this broadcast is received.  
 	**>
-	
 	broadcast AppsInfoUpdated {}	
-	
-	
+
 	<**
 		@description : AppInfo is a map of manifest data. It stores Key and value as strings .The manifest should be a part of app bundle and app manager extracts and stores this data
 	**>
-	map AppInfo
+	struct AppInfo
 	{
-		String to String
+                String app_id
```

Comment:
My idea is that you can find your way back using the info. This way you have a single object representing the app information that you can use to do other calls regarding the app. Otherwise you need to keep track of which info belongs to which app. But you are right, that could be done client side.